### PR TITLE
IGAPP-97: Use webkit on iOS

### PR DESCRIPTION
**Pull request by Schedulaar**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

It seems Pois.js and PoisContainer.js had the wrong line-endings. I ended up converting them to LF. Not sure though, what happened exactly :see_no_evil: 

Also this has to be tested on iOS (maybe one needs to do `pod install` as wel...)

### CODE REVIEW

File: native/src/modules/common/components/RemoteContent.js [hunk 1]
```diff
@@ -73,17 +69,14 @@ class RemoteContent extends React.Component<PropType, StateType> {
...
     const width = this.state.webViewWidth
     return <StyledView onLayout={this.onLayout}>
       <WebView
-        source={createHtmlSource(renderHtml(content, files, theme, language),
-          URL_PREFIX + getResourceCacheFilesDirPath(cityCode))}
-        allowFileAccess // Needed by android to access file:// urls
+        source={createHtmlSource(renderHtml(content, files, theme, language, resourceCacheUrl), resourceCacheUrl)}
         originWhitelist={['*']} // Needed by iOS to load the initial html
         javaScriptEnabled
-        useWebKit={false}
```
Comment (klinzo) [inline native/src/modules/common/components/RemoteContent.js:86]: 🎉

File: native/src/modules/common/renderHtml.js [hunk 2]
```diff
@@ -41,28 +43,28 @@ const renderJS = (files: PageResourceCacheStateType) => `
...
   var hrefs = document.querySelectorAll('[href]')
   var srcs = document.querySelectorAll('[src]')
-  var files = ${JSON.stringify(files)}
-  
-  console.debug('Files to inject:') // TODO: remove
-  console.debug(files)
-  
+  var cacheDictionary = ${JSON.stringify(cacheDictionary)}
+
+  console.debug('Resources to inject:') // TODO: remove
```
Comment (klinzo) [inline native/src/modules/common/renderHtml.js:48]: I think the TODO could be removed

File: native/src/routes/disclaimer/Disclaimer.js [hunk 3]
```diff
@@ -25,7 +25,7 @@ class Disclaimer extends React.Component<PropsType> {
                  navigation={navigation}
                  files={{}}
                  language={language}
-                 cityCode={city}
+                 resourceCacheUrl='http://localhost:8080'
```
Comment (klinzo) [inline native/src/routes/disclaimer/Disclaimer.js:28]: Why did you choose to hardcode it here? Shouldn't you rather pass nothing? Just seems weird.

File: native/src/modules/static-server/containers/StaticServerProvider.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      this.setState({ errorMessage: e.message })
+    }
+  }
+
+  componentWillUnmount () {
+    this.staticServer.stop()
+  }
+
+  render () {
+    if (this.state.errorMessage != null) {
```
Comment (klinzo) [inline native/src/modules/static-server/containers/StaticServerProvider.js:48]:
```suggestion
    if (this.state.errorMessage !== null) {
```

**Review by klinzo — CHANGES_REQUESTED** [review]
If I am clicking on a leaf category, safari opens localhost:8080 🤔 

Happens also for events, pois and disclaimer.

**Comment by Schedulaar** [comment]
@klinzo Ahh.. Yeah, fortunately, I know why this happens 🙈 I just forgot to readd a line...
Could you test it on a real iOS device again, please? (Simulator never was a problem)

**Comment by klinzo** [comment]
AFAICT it still opens the browser (localhost) :/

**Review by klinzo — APPROVED** [review]

**Review by Taggotty — APPROVED** [review]
Looks alright to me, although this is definitely not my field of expertise ^^

File: native/src/modules/common/components/RemoteContent.js [hunk 5]
```diff
@@ -58,9 +58,9 @@ class RemoteContent extends React.Component<PropType, StateType> {
   }
 
   onShouldStartLoadWithRequest = (event: WebViewNavigation) => {
-    const url = event.url
+    const url = event.url.endsWith('/') ? event.url.slice(0, -1) : event.url
```
[the remark this answers is not in the document]
Comment (klinzo) [reply native/src/modules/common/components/RemoteContent.js:61]: resourceCachUrl is `https://localhost:8080`, so if there is a query the check fails anyway, so I guess this should be fine. Right @Schedulaar ?

[the remark this answers is not in the document]
Comment (Schedulaar) [reply native/src/modules/common/components/RemoteContent.js:61]: Yeah, but we should probably only use the normalized url for comparing below.
We should still use `this.props.onLinkPress(event.url)` imo

**Review by maxammann — APPROVED** [review]

**Review by maxammann — APPROVED** [review]

**Pull request merged**